File: 1-two-sum/1-two-sum.cpp
class Solution
{
  public:
    vector<int> twoSum(vector<int>& nums, int target)
    {
        unordered_map<int, int> hash;
        for(int i=0; i<nums.size(); i++)
        {
            int n = nums[i], m = target-n;
            if(hash.find(m) != hash.end())
                return {hash[m], i};
            else
                hash[n] = i;
        }
        return {};
    }
};

File: 1-two-sum/1-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        hash = {}
        for i, n in enumerate(nums):
            m = target - n
            if m in hash:
                return [hash[m], i]
            else:
                hash[n] = i